src/app/reporting/reporting.cpp

```diff
@@ -29,18 +29,23 @@ void MatterReportingAttributeChangeCallback(EndpointId endpoint, ClusterId clust
     // Attribute writes have asserted this already, but this assert should catch
     // applications notifying about changes from their end.
     assertChipStackLockedByCurrentThread();
-
-    InteractionModelEngine::GetInstance()->GetDataModelProvider()->Temporary_ReportAttributeChanged(
-        AttributePathParams(endpoint, clusterId, attributeId));
+    auto * engine = InteractionModelEngine::GetInstance();
+    VerifyOrReturn(engine);
```

Comment:
Why is anyone calling this before data model has been initialized?  That indicates a serious bug, and papering over it like this will just lead to wrong behavior as far as I can tell....

Reply:
In my test, `InitChipStack()` is called before Server instance is initialized. And before server instance is initialized, the connectivityMgr(this is initialized with the chip stack) might receive a Wi-Fi connect event, which will lead to that the Wi-Fi driver tries to call MatterReportingAttributeChangeCallback() to report the network status change. If the server instance  doesn't set the data model provider for IM engine at that, there will be crash.

Reply:
I see, so this is a case where the attribute value is not owned by the data model at all, so it's ok if it's changing before the data model is set up?

Reply:
The network status is read from the network driver so that it might be changed before data model is set up. I raised another PR which does the check before at where the network status is reported.